bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -144,25 +146,32 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		return ctrl.Result{}, err
 	}
 
+	// Initialize the patch helper.
+	patchHelper, err := patch.NewHelper(config, r.Client)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
 	switch {
-	// Wait patiently for the infrastructure to be ready
+	// Wait for the infrastructure to be ready.
 	case !cluster.Status.InfrastructureReady:
 		log.Info("Infrastructure is not ready, waiting until ready.")
 		return ctrl.Result{}, nil
-	// bail super early if it's already ready
+	// Migrate plaintext data to secret.
+	case config.Status.BootstrapData != nil:
+		if err := r.storeBootstrapData(ctx, config, config.Status.BootstrapData); err != nil {
+			return ctrl.Result{}, err
+		}
+		config.Status.BootstrapData = nil
+		return ctrl.Result{}, patchHelper.Patch(ctx, config)
+	// Return early if the configuration and Machine's infrastructure are ready.
 	case config.Status.Ready && machine.Status.InfrastructureReady:
-		log.Info("ignoring config for an already ready machine")
+		log.Info("Ignoring config for a ready machine")
 		return ctrl.Result{}, nil
 	// Reconcile status for machines that have already copied bootstrap data
```

Comment:
Remind me when this scenario is possible?

Reply:
After pivot